### PR TITLE
chore(TCOMP-2148) - sample connectors

### PR DESCRIPTION
**Pull request by acatoire**
### Why this PR is needed?
To improve the TCK API test capabilities, we needed a sample connector plugin with all the TCK elements implemented.

### What does this PR adds (design/code thoughts)?
- a sample connector plugin for API test purpose.
- a new endpoint test set in talend-component-maven-plugin/src/it/web/test/tcomp_endpoints.json
- a new manual launcher for test server in talend-component-maven-plugin/src/it/web/.jenkins/scripts/server-registry-manual_start.sh

### CODE REVIEW

File: .jenkins/settings.xml [hunk 1]
```diff
@@ -81,6 +81,22 @@
...
+    <server>
+      <id>snapshots</id>
+      <username>${env.NEXUS_USER}</username>
+      <password>${env.NEXUS_PASSWORD}</password>
+      <configuration>
+        <timeout>30000</timeout>
+      </configuration>
+    </server>
+    <server>
+      <id>releases</id>
```
Comment (undx) [inline .jenkins/settings.xml:93]: same comment as above...

File: pom.xml [hunk 2]
```diff
@@ -1228,6 +1228,7 @@
           <consoleOutput>true</consoleOutput>
           <reportFile>${project.build.directory}/${project.build.finalName}.rat</reportFile>
           <excludes>
+            <exclude>**/connectors/**/*.properties</exclude>
```
Comment (undx) [inline pom.xml:1231]: Why exclude this module?  Just fill-in correct licenses in mentioned files during build.

File: sample-parent/demo-connector/src/main/java/org/talend/sdk/component/test/connectors/service/UIService.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public SuggestionValues getListEntities() {
+        final List<Item> entities =
+                Arrays.stream(new String[] { "Account", "Customer", "Prospect", "Quotation", "Sale" })
+                        .map(e -> new Item(e, e))
+                        .collect(Collectors.toList());
+
+        return new SuggestionValues(true, entities);
+    }
+
+    @Update("UPDATE_DUMMYSUB")
```
Comment (undx) [inline sample-parent/demo-connector/src/main/java/org/talend/sdk/component/test/connectors/service/UIService.java:49]: make a const like above...

File: sample-parent/demo-connector/src/main/java/org/talend/sdk/component/test/connectors/config/InputConfig.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+public class InputConfig implements Serializable {
+
+    public final static int INPUT_CONFIG_VERSION = 3;
+
+    @Option
+    @Documentation("The dataset.")
+    private Dataset dataset;
+
+    @Option
+    @Documentation("Retrieve only dataafter the given date.")
```
Comment (undx) [inline sample-parent/demo-connector/src/main/java/org/talend/sdk/component/test/connectors/config/InputConfig.java:40]: typo

**Comment by undx** [comment]
When pr will be accepted, do plz, a squash merge !!!

File: sample-parent/demo-connector/src/main/resources/org/talend/sdk/component/test/connectors/config/Messages.properties [hunk 5]
```diff
@@ -0,0 +1,23 @@
+Authent.login._displayName = Login
+Authent.login._placeholder = Login placeholder
+Authent.nbAuthentRetry._displayName = Number of authentication retry
+Authent.password._displayName = Password
+Authent.password._placeholder = Password placeholder
+Dataset.datastore._displayName =
```
[the remark this answers is not in the document]
Comment (acatoire) [reply sample-parent/demo-connector/src/main/resources/org/talend/sdk/component/test/connectors/config/Messages.properties]: Corrected

File: sample-parent/demo-connector/src/main/java/org/talend/sdk/component/test/connectors/config/DummySub.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import org.talend.sdk.component.api.configuration.Option;
+import org.talend.sdk.component.api.configuration.ui.layout.GridLayout;
+import org.talend.sdk.component.api.meta.Documentation;
+
+import lombok.Data;
+
+@Data
+@Documentation("Dummy configuration.")
+@GridLayout({ @GridLayout.Row({ "abc" }), @GridLayout.Row({ "def" }) })
+public class DummySub {
```
[the remark this answers is not in the document]
Comment (acatoire) [reply sample-parent/demo-connector/src/main/java/org/talend/sdk/component/test/connectors/config/DummySub.java]: All been correcter to demo or other

File: sample-parent/demo-connector/src/main/java/org/talend/sdk/component/test/connectors/input/DummyInput.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    public void init() {
+    }
+
+    @PreDestroy
+    public void release() {
+    }
+
+    @Producer
+    public Object next() {
+        return null;
```
[the remark this answers is not in the document]
Comment (acatoire) [reply sample-parent/demo-connector/src/main/java/org/talend/sdk/component/test/connectors/input/DummyInput.java]: It now returns a LocalDate.now();

File: sample-parent/sample-connector/README.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# Demo Connector
+
+## Description
+This module contains a demo connector developed as a QA tool.
```
Comment (fdesiles) [inline sample-parent/sample-connector/README.md:4]:
```suggestion
This module contains a sample (demonstration) connector developped as a QA tool.
```

File: sample-parent/sample-connector/README.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+## Goal
+This demo connector has been firstly built to cover endpoint tests with API Tester.
+You can see actual tests in [API test project](../../talend-component-maven-plugin/src/it/web/README.md)
+It should have enough content to be able to validate all tck api endpoints with all possible parameters.
+The connectors should have everything translated in at least 2 languages, no need for real translation
+
+# Translation
+In order to have an easy testing of language mechanic, a simple python script has been provided in the project: [i18n_messages_properties_generator.py](src/main/resources/org/talend/sdk/component/test/connectors/i18n_messages_properties_generator.py).
+Simply execute it to generate a copy of every `messages.properties` file in multiples languages.
```
Comment (fdesiles) [inline sample-parent/sample-connector/README.md:15]:
```suggestion
Simply execute it to generate a copy of every `messages.properties` files in multiples languages.
```

File: talend-component-maven-plugin/src/it/web/README.md [hunk 10]
```diff
@@ -10,29 +10,31 @@ You will find:
...
-## Execution
+### Start local server
 The execution with maven plugin is done in several steps as follows:
-- Go into the tested connector folder (TODO: use the new demo component)
-- Build the connector
-- Start the tcomp server using the component-tool-webapp 
-  - Use Dtalend.web.batch, Dtalend.web.batch.timeout, Dtalend.web.port options depending on your needs (default test environment is (https://localhost:8080))
-- Go to the test location folder (/talend-component-maven-plugin/src/it/web/test)
-- Run the tests from chosen file with your Talend instance and account ID
+- Go into the demo connector folder and build it with Maven.
```
Comment (fdesiles) [inline talend-component-maven-plugin/src/it/web/README.md:23]:
```suggestion
- Go into the sample connector folder and build it with Maven.
```

File: talend-component-maven-plugin/src/it/web/README.md [hunk 11]
```diff
@@ -45,12 +47,13 @@ If you want to edit the test or simply play them in API Tester environment, you
...
-- Request to be part of the connectors group in rd.aws.eu.talend.com Tenant
-
-If you are not part of Talend
-- Import the test project form (/talend-component-maven-plugin/src/it/web/test) in your API Tester DRIVE
-
+- Import the test project that you want to work on, in API Tester
+  - from: `/talend-component-maven-plugin/src/it/web/test`.
+- It will be loaded on your personnal API Tester DRIVE.
+- You can then edit and test your modifications directly on API Tester while running [local server](#Start-local-server)
+- When the edition is finish, you can push it back to GitHub branch and make a Push Request.
```
Comment (fdesiles) [inline talend-component-maven-plugin/src/it/web/README.md:54]:
```suggestion
- When the edition is finish, you can push it back to GitHub branch and make a Pull Request.
```

**Review by fdesiles — CHANGES_REQUESTED** [review]

**Review by ypiel-talend — APPROVED** [review]
LGTM

**Review by fdesiles — APPROVED** [review]

File: sample-parent/sample-connector/src/main/resources/icons/README.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+Downloaded from
+https://icons8.com/icon/set/plants/material
+Or created with 
+https://uxwing.com/svg-icon-editor/
```
Comment (undx) [inline sample-parent/sample-connector/src/main/resources/icons/README.md:4]: Are those ressources OSS? Need to indicate license...

File: talend-component-maven-plugin/src/it/web/.jenkins/Jenkinsfile [hunk 13]
```diff
@@ -24,10 +24,9 @@
 /**
  * Pod configuration
  */
+// TODO move to custom builder
 final String _TSBI_IMAGE = 'jdk11-svc-springboot-builder'
 final String _TSBI_VERSION = '2.9.18-2.4-20220104141654'
```
Comment (undx) [inline talend-component-maven-plugin/src/it/web/.jenkins/Jenkinsfile:29]: Should be sync to master.

**Review by undx — COMMENTED** [review]
See my comments. Also,
- Connectors have a strange class names (TheInput1 for instance), maybe remove the _1_.
- Is that normal the we mix the sample-connector and the IT for testing endpoints, isn't there another issue for it? 
- remove empty shell files, they're useless.

File: sample-parent/sample-connector/pom.xml [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+  <artifactId>sample-connector</artifactId>
+
+  <name>Component Runtime :: Sample Parent :: Sample connectors</name>
+  <description>Sample connectors for test.
+    This connector plugin include several connectors used for learning process and automation testing.
+    Examples of automation testing projects:
+     - TCK API automatic testing.</description>
+
+  <properties>
+    <talend.build.name>${talend.build.name.base}.sample</talend.build.name>
```
[the remark this answers is not in the document]
Comment (acatoire) [reply sample-parent/sample-connector/pom.xml:35]: corrected

File: sample-parent/sample-connector/src/main/resources/icons/README.md (hunk 12 above)
Comment (acatoire) [reply sample-parent/sample-connector/src/main/resources/icons/README.md:4]: corrected

File: sample-parent/sample-connector/src/main/java/org/talend/sdk/component/test/connectors/package-info.java [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@Components(
+        family = "the_family",
+        categories = "the_categorie_1")
```
[the remark this answers is not in the document]
Comment (acatoire) [reply sample-parent/sample-connector/src/main/java/org/talend/sdk/component/test/connectors/package-info.java:18]: corrected

**Comment by acatoire** [comment]
> Is that normal the we mix the sample-connector and the IT for testing endpoints, isn't there another issue for it?
Agreed, I split this and made a new branch/PR for it: https://github.com/Talend/component-runtime/pull/696

**Review by undx — APPROVED** [review]

**Pull request merged**